HealthcareSystem/GraphicalEditor/Models/MapObjectRelated/IdGenerator.cs

```diff
@@ -12,12 +12,12 @@ public  class IdGenerator
 
         public IdGenerator()
         {
-            _id = GetNextID();
+            _id = 0;
         }
 
         public long GetNextID()
         {
-            MainWindow wnd = (MainWindow)App.Current.MainWindow;
+            MainWindow wnd = new MainWindow();
```

Comment:
I assume that this is a patch that happened because when you link to the other WPF project, the main window is no longer the one from the graphical editor. Creating a new instance means that this will not count the objects correctly at all, and you will get the same id every time. The MainWindow dependency should be removed.